pandas/core/reshape/merge.py

```diff
@@ -1138,6 +1141,88 @@ def _get_join_indexers(left_keys, right_keys, sort=False, how='inner',
     return join_func(lkey, rkey, count, **kwargs)
 
 
+def _restore_dropped_levels_multijoin(left, right, dropped_level_names,
+                                      join_idx, lidx, ridx):
+    """
+    *this is an internal non-public method*
+
+    Returns the levels, labels and names of a multil-index to multi-index join.
+    Depending on the type of join, this method restores the appropriate
+    dropped levels of the joined multi-index. The method relies on lidx, ridx
+    which hold the index positions of left and right, where a join was feasible
+
+    Parameters
+    ----------
+    left : Index
```

Comment:
These are always MultiIndexes, correct? If so let's specify that.

Also, to shorten these a bit, numpydoc allows you to group similar parameters

```
left, right : MultiIndex
    The original left and right indexes
...
```

could do the same for `lidx` and `ridx`

Reply:
can you update this